src/provider/mod.rs

```diff
@@ -79,20 +81,67 @@ pub struct Provider {
     pub oauth2_authorizer: Option<Oauth2Authorizer>,
 }
 
-pub fn get_provider_info(addr: &str) -> Option<&Provider> {
-    let domain = match addr.parse::<EmailAddress>() {
-        Ok(addr) => addr.domain,
-        Err(_err) => return None,
+pub async fn get_provider_info(domain: &str) -> Option<&'static Provider> {
```

Comment:
Now `get_provider_info()` takes a domain instead of an addr, which is a breaking change, which we should probably avoid (OTOH our Rust API is not widely used yet).

The ffi call in `lib.rs` would have to be adapted, too, though, if we keep this behavior (currently every call to `dc_provider_new_from_email()` would just fail as an address is given but a domain is expected). But as the ffi should not do any logic, just convert from C to Rust, I don't actually think it should be adapted.

Maybe just see if there's an "@" in there and if yes, treat as email address, otherwise treat as domain?

Reply:
I added support for email addresses for compatibility and a test.